python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/launcher_tests/test_launching.py

```diff
@@ -100,6 +100,93 @@ def test_default_launcher(
     ecs.stop_task(task=task_arn)
 
 
+def test_launcher_dont_use_task_definition(
+    ecs,
+    instance_dont_use_current_task_definition,
+    workspace,
+    external_pipeline,
+    pipeline,
+    subnet,
+    image,
+    environment,
+):
+    instance = instance_dont_use_current_task_definition
+    run = instance.create_run_for_pipeline(
+        pipeline,
+        external_pipeline_origin=external_pipeline.get_external_origin(),
+        pipeline_code_origin=external_pipeline.get_python_origin(),
+    )
+
+    cluster = instance.run_launcher.cluster
+
+    assert not run.tags
+
+    initial_task_definitions = ecs.list_task_definitions()["taskDefinitionArns"]
+    initial_tasks = ecs.list_tasks(cluster=cluster)["taskArns"]
+
+    instance.launch_run(run.run_id, workspace)
+
+    # A new task definition is created
+    task_definitions = ecs.list_task_definitions()["taskDefinitionArns"]
+    assert len(task_definitions) == len(initial_task_definitions) + 1
+    task_definition_arn = list(set(task_definitions).difference(initial_task_definitions))[0]
+    task_definition = ecs.describe_task_definition(taskDefinition=task_definition_arn)
+    task_definition = task_definition["taskDefinition"]
+
+    assert instance.run_launcher.check_run_worker_health(run).status == WorkerStatus.RUNNING
+
+    # It has a new family, name, and image
+    # We get the family name from the location name. With the InProcessExecutor that we use in tests,
+    # the location name is always <<in_process>>. And we sanitize it so it's compatible with the ECS API.
+    assert task_definition["family"] == "in_process"
+    assert len(task_definition["containerDefinitions"]) == 1
+    container_definition = task_definition["containerDefinitions"][0]
+    assert container_definition["name"] == "run"
+    assert container_definition["image"] == image
+    assert not container_definition.get("entryPoint")
+    assert not container_definition.get("dependsOn")
+    # But other stuff is inherited from the parent task definition
```

Comment:
I know this is largely copy-pasted from other tests, but isn't this the exact opposite of what we're trying to support here?

Can we maybe reduce the test assertions to just the behavior that diverges from the default?